golos.publication/parameters.hpp

```diff
@@ -0,0 +1,69 @@
+#pragma once
+#include <eosiolib/singleton.hpp>
+#include <common/parameter.hpp>
+#include <common/parameter_ops.hpp>
+
+namespace golos {
+
+    using namespace eosio;
+
+    struct st_max_vote_changes : parameter {
+        uint8_t max_vote_changes;
+
+        void validate() const override {
+            eosio_assert(max_vote_changes > 0, "Max vote changes can't be negative.");
+        }
+    };
+    using max_vote_changes_prm = param_wrapper<st_max_vote_changes, 1>;
+
+    struct st_cashout_window : parameter {
+        uint32_t cashout_window;
+
+        void validate() const override {
+            eosio_assert(cashout_window > 0, "Cashout window can't be negative.");
+        }
+    };
+    using cashout_window_prm = param_wrapper<st_cashout_window, 1>;
+
+    struct st_upvote_lockout : parameter {
+        uint8_t upvote_lockout;
+        
+        void validate() const override {
```

Comment:
remove `validate()`, there is no restrictions. the only restriction is that `upvote_lockout` must be less then `cashout_window` (to allow upvotes), but it must be checked in setter visitor. example is in ctrl: https://github.com/GolosChain/golos-smart/blob/635e982d5dbf3b3939797d5c479b46538ba869f9/golos.ctrl/src/golos.ctrl.cpp#L34

actually, we must check this in both `upvote_lockout` and `cashout_window` setters, because either can make other invalid. So here is parameters dependance. Propose combine both this parameters into one with 2 fields, so we can check values at it's validator.